### ISSUE TITLE
scss overlay for ng form controls

### ISSUE THREAD

**Issue by lucienbertin** [LuccaSA/lucca-front#21]
angular uses classes like `ng-valid`, `ng-invalid`, `ng-pristine` etc... it would be super usefull if the scss framework offered an option to have style automatically added to said classes

https://github.com/LuccaSA/lucca-front/blob/master/packages/ng/libraries/core/src/style/components/_input.scss

**Comment by lucienbertin** [comment]
@jeremie-lucca the code associated must be in `/ng`, not in the `/scss` package

**Comment by lucienbertin** [comment]
@jeremie-lucca you might want to use placeholders instead of mixins for this

[placeholders vs mixins](https://www.sitepoint.com/sass-mixin-placeholder/)

by using placeholders you can write

```scss
%input-invalid {
  border: 1px solid red;
}
.is-invalid {
  @extend %input-invalid;
}
.ng-invalid {
  @extend %input-invalid;
}
```
and the transpiled code will be
```css
.is-invalid, .ng-invalid {
  border: 1px solid red;
}
```

**Comment by jeremie-lucca** [comment]
That's a great idea, so I tried switching mixins for placeholders, but I encountered an issue by making a try on input.is-error:

With mixin, the CSS will happen on the input.is-error place.
With placeholder, the CSS will happen on the placeholder place. 

It leads to switch cascading order between is-error and :focus, so focus override is-error.
I prefere staying with mixins to keep the specificity low.
